### ISSUE TITLE
Serializing a surrogate pair causes an IllegalArgumentException

### ISSUE THREAD

**Issue by Shamicen** [pdvrieze/xmlutil#113]
Serializing a string containing a Surrogate pair, for example: 🔒 ``("\uD83D\uDD12")``, will result in the following Exception to be thrown:
```
java.lang.IllegalArgumentException: In xml 1.0 the character 0xd83d is not valid
nl.adaptivity.xmlutil.core.KtXmlWriter.appendXmlChar$throwInvalid(KtXmlWriter.kt:127)
nl.adaptivity.xmlutil.core.KtXmlWriter.appendXmlChar(KtXmlWriter.kt:152)
nl.adaptivity.xmlutil.core.KtXmlWriter.writeEscapedText(KtXmlWriter.kt:160)
nl.adaptivity.xmlutil.core.KtXmlWriter.text(KtXmlWriter.kt:356)
nl.adaptivity.xmlutil.serialization.XmlEncoderBase$TagEncoder$encodeStringElement$2.invoke(XMLEncoder.kt:558)
nl.adaptivity.xmlutil.serialization.XmlEncoderBase$TagEncoder$encodeStringElement$2.invoke(XMLEncoder.kt:557)
nl.adaptivity.xmlutil.serialization.XmlEncoderBase$TagEncoder.endStructure(XMLEncoder.kt:568)
```
The problem seems to be that the illegal character check [here](https://github.com/pdvrieze/xmlutil/blob/672d85a61314957126502db639902b3a59648e6b/core/src/commonMain/kotlin/nl/adaptivity/xmlutil/core/KtXmlWriter.kt#L152) does not account for Surrogate pairs.

**Comment by pdvrieze** [comment]
This encoder is based upon the Android XML encoder. Surrogate pairs cannot be encoded as pairs, but need to be encoded as unicode code points (using separate character entities is invalid). To address this the code needs to be updated to work with code points instead.

**Comment by pdvrieze** [comment]
Unfortunately codepoints are not supported on multiplatform so I'll have to work around that manually.

**Comment by pdvrieze** [comment]
I've updated this in dev. The push should automatically trigger a snapshot release.

**Comment by pdvrieze** [comment]
Now released in [0.85.0](https://github.com/pdvrieze/xmlutil/releases/tag/v0.85.0).
